base/dataset.go

```diff
@@ -35,27 +37,40 @@ func OpenDataset(ctx context.Context, fsys qfs.Filesystem, ds *dataset.Dataset)
 			return
 		}
 	}
+
 	if ds.Readme != nil && ds.Readme.ScriptFile() == nil {
-		if err = ds.Readme.OpenScriptFile(ctx, fsys); err != nil {
-			log.Debug(err)
-			return
+		readmeTimeoutCtx, cancel := context.WithTimeout(ctx, time.Millisecond*250)
```

Comment:
This seems error prone? I'd rather have this behavior within the `OpenScriptFile` or specifically in the `qfs.PathResolver` that actually implements this. On the `http` resolver it might make sense to have a timeout + retry mechanics etc. while on the `local` version this should definitelly get back in < 250ms.

Reply:
Definitely agree with this long term. mind opening an issue on qri-io/qfs?